sklearn/calibration.py

```diff
@@ -942,6 +1107,67 @@ def predict(self, T):
         return expit(-(self.a_ * T + self.b_))
 
 
+class _TemperatureScaling(RegressorMixin, BaseEstimator):
+    """Temperature scaling model.
+
+    Attributes
+    ----------
+    T_initial_ : float
+        The initial temperature for optimization.
+
+    T_ : float
+        The optimized temperature.
+    """
+
+    def __init__(self, T_initial=1.0):
+        self.T_initial_ = T_initial
+        self._beta_0 = np.divide(1.0, self.T_initial_)
```

Comment:
The constructor of scikit-learn estimators should always store the input parameters unchanged with the same names:

https://scikit-learn.org/stable/developers/develop.html#instantiation